# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: xpcom/base/nsIMemoryReporter.idl
# Commit: 9fccb38453a4
# Full Hash: 9fccb38453a4fc43b21784c850d4236fdea7d489
# Author: Gabriele Svelto <gsvelto@mozilla.com>
# Date: 2019-11-02 21:51:29
# Regressor Bug: 1586236
# File Overlap Count: 1
# Description:
#   Bug 1586236 - Use memory resource notifications to detect low memory scenarios on Windows; r=dmajor
#   
#   This patch uses the low memory resource notification facility to detect
#   scenarios where physical memory is running low without polling. This is a
#   significant change compared to the previous behavior which measured both
# ==============================================================================

diff -r d60821c527f2 -r 9fccb38453a4 xpcom/base/nsIMemoryReporter.idl
--- a/xpcom/base/nsIMemoryReporter.idl	Sat Nov 02 08:19:46 2019 +0000
+++ b/xpcom/base/nsIMemoryReporter.idl	Fri Nov 01 23:08:59 2019 +0000
@@ -373,9 +373,9 @@
    *
    * |storageSQLite| (UNITS_BYTES)  Memory used by SQLite.
    *
-   * |lowMemoryEvents{Virtual,CommitSpace,Physical}| (UNITS_COUNT_CUMULATIVE)
-   * The number of low-{virtual,commit-space,physical}-memory events that have
-   * occurred since the process started.
+   * |lowMemoryEventsPhysical| (UNITS_COUNT_CUMULATIVE)
+   * The number of low-physical-memory events that have occurred since the
+   * process started.
    *
    * |ghostWindows| (UNITS_COUNT)  A cached value of the number of ghost
    * windows. This should have been updated within the past 60s.
@@ -404,8 +404,6 @@
 
   [must_use] readonly attribute int64_t storageSQLite;
 
-  [must_use] readonly attribute int64_t lowMemoryEventsVirtual;
-  [must_use] readonly attribute int64_t lowMemoryEventsCommitSpace;
   [must_use] readonly attribute int64_t lowMemoryEventsPhysical;
 
   [must_use] readonly attribute int64_t ghostWindows;
@@ -505,8 +503,6 @@
 DECL_REGISTER_DISTINGUISHED_AMOUNT(Infallible, StorageSQLite)
 DECL_UNREGISTER_DISTINGUISHED_AMOUNT(StorageSQLite)
 
-DECL_REGISTER_DISTINGUISHED_AMOUNT(Infallible, LowMemoryEventsVirtual)
-DECL_REGISTER_DISTINGUISHED_AMOUNT(Infallible, LowMemoryEventsCommitSpace)
 DECL_REGISTER_DISTINGUISHED_AMOUNT(Infallible, LowMemoryEventsPhysical)
 
 DECL_REGISTER_DISTINGUISHED_AMOUNT(Infallible, GhostWindows)